# DP: Fix basic HTTP authentication in urllib2.py (taken from 2.7).

--- a/Lib/urllib2.py
+++ b/Lib/urllib2.py
@@ -839,7 +839,10 @@
             if mo:
                 scheme, quote, realm = mo.groups()
                 if scheme.lower() == 'basic':
-                    return self.retry_http_basic_auth(host, req, realm)
+                    response = self.retry_http_basic_auth(host, req, realm)
+                    if response and response.code != 401:
+                        self.retried = 0
+                    return response

     def retry_http_basic_auth(self, host, req, realm):
         user, pw = self.passwd.find_user_password(realm, host)

